americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+        kwargs['xlabel'] = 'Reference Group'
+    if 'xlim' not in kwargs:
+        kwargs['xlim'] = xlim
+
+    # Formats the axis, to make it pretty
+    _format_axis(ax, **kwargs)
+
+    # Shows the seperation line if desired
+    if show_seperation:
+        xlim = ax.get_xlim()
+        lower_y, upper_y = ax.get_ylim()
+        tick_dist = (upper_y - lower_y)/5.
+        ax.plot(np.arange(-0.25, 0.75, 0.01) *
+                (1 + np.floor(len(kwargs['xticks']))),
+                np.array([-tick_dist*sep_size, tick_dist*sep_size]*50) +
+                lower_y + tick_dist*0.4,
+                'k-', linewidth=7)
+        ax.plot(np.arange(-0.25, 0.75, 0.01) *
+                (1 + np.floor(len(kwargs['xticks']))),
+                np.array([-tick_dist*sep_size, tick_dist*sep_size]*50) +
+                lower_y + tick_dist*0.4,
+                'w-', linewidth=3)
+    else:
+        xlim = ax.get_xlim()
+
+    # Checks the format, just to be sure... Formats the axis, to make it pretty
+    ax.set_xlim(xlim)
+
+    # Formats the yaxis to show seperation
+    if show_seperation:
+        yticklabels = ax.get_yticks()
+        yticklabels[0] = 0
+        ax.set_yticklabels(yticklabels, size=kwargs['ytick_size'])
+
+    # Adds a legend to the top to distinguish groups
+    # Determines the horizontal ordering for the groups
+    num_groups = len(order)
+    num_leg_cols = int(np.ceil(float(num_groups)/2))
+    leg_order = np.hstack([np.array([i, i+num_leg_cols])
+                           for i in xrange(num_leg_cols)])[:num_groups]
+    # Gets the patches and labels
+    patches = np.array(ax.patches[:num_groups])[leg_order]
+    labels = np.array([l.split('(')[0].replace('_', ' ')
+                       for l in order])[leg_order]
+    # Adds the legend
+    ax.legend(patches, labels,
+              loc="upper center",
+              ncol=num_leg_cols,
+              fontsize=12,
+              frameon=False)
+
+    return ax
+
+
+def split_taxa(taxa, level=7):
+    """Splits a greengenes taxonomy string into a dataframe
+
+    Parameters
+    ----------
+    taxa : list
+        a list of greengenes strings
+    level: {2, 3, 4, 5, 6, 7}
+        the taxonomic level for summary
+
+    Returns
+    -------
+    splits : array
+        the taxonomy data summarized to the specified level. If data is missing
+        for a particular level, the string will specify the last known string.
+        (i.e. k__kingdom;p__phylum;c__pclass;o__porder;f__family;g__;s__
+        outputs with a genus and species value "f. family").
+
+    levels: list
+        the name of the taxonomic levels corresponding to columns in the table.
+    """
+
+    if level not in {2, 3, 4, 5, 6, 7}:
+        raise ValueError('%r is not a supported taxonomic level.' % level)
+
+    # Prealocates a holding object
+    splits = np.zeros((1, (level)))
+
+    # Sets up the names of phylogenetic levels
+    levels = ['kingdom', 'phylum', 'p_class', 'p_order', 'family', 'genus',
```

Comment:
i dont see that happening anywhere in the code, and they are not reserved strings